02_bubble/brvishnu92/main.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(s []int) []int {
+	sort := make([]int, len(s))
+	copy(sort, s)
+	swap := true
+	for swap {
+		z := sort[0]
+		swap = false
+		for i, x := range sort {
+			if x < z && i != 0 {
+				swap = true
+				sort[i-1], sort[i] = sort[i], sort[i-1]
+			}
+			z = x
+		}
+	}
+	return sort
+}
+
+func insertion(s []int) []int {
+	sort := make([]int, len(s))
+	copy(sort, s)
+	for i := 1; i < len(sort); i++ {
+		j := i
```

Comment:
I believe you should be able to write the inner loop like a standard `for` loop as you have done above.

Reply:
I have assumed u meant the for outer for loop and updated the code. As the inner for loop just checks the index. Hope its what you meant.

Reply:
Sorry I wasn't not very clear in my description.  What I meant was that you probably do not need to initialise and increment the variables separately.  You can do it as part of the for loop.

```go
for j := i; j > 0; j-- {
	if sort[j-1] > sort[j] {
		sort[j-1], sort[j] = sort[j], sort[j-1]
	}
}
```

I think that will achieve the same thing

Reply:
Oh got it :-)  Thanks